### PR TITLE
Count vowels rumiya

### PR DESCRIPTION
**Pull request by Ismatova-Rumiya**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [ ] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [ ] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [x] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [ ] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [x] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [x] The function has correct type annotations

### Do's

- [x] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [x] The function's strategy _is not_ described in the documentation
- [x] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [x] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [x] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [ ] The code includes defensive assertions
- [ ] Defensive assertions include as little logic as possible

### CODE REVIEW

**Comment by JawidMohseni** [comment]
I am reviewing it

**Review by JawidMohseni — CHANGES_REQUESTED** [review]
The implementation is solid, with the function and tests written clearly. However, there are a few missing elements and improvements needed to fully meet the checklist standards.
please see the comment and check if it can be change as it expected

**Comment by Ismatova-Rumiya** [comment]
I have edited the code and the unit test taking the comments into consideration

File: solutions/count_vowels.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+Created on 2026-07-01
+Author: Rumiya Ismatova
+"""
+
+
+def count_vowels(text: str) -> int:
+    """
+    Count_vowels function takes a string and count
+    all upper and lower case vowels.
+
```
[the remark this answers is not in the document]
Comment (Ismatova-Rumiya) [reply solutions/count_vowels.py:18]: Done!

File: solutions/tests/test_count_vowels.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+Created on 2025-01-07
+
+@author: Rumiya Ismatova
+"""
+
+import unittest
+from solutions.count_vowels import count_vowels
+
+
+class TestCountVowels(unittest.TestCase):
```
[the remark this answers is not in the document]
Comment (Ismatova-Rumiya) [reply solutions/tests/test_count_vowels.py:15]: Done!

File: solutions/tests/test_count_vowels.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    def test_numbers_and_symbols(self):
+        """
+        Test that a string with numbers and symbols returns 0 vowels.
+        Ensures non-alphabetic characters are ignored.
+        """
+        self.assertEqual(count_vowels("12345!@#$%"), 0)
+
+
+if __name__ == "__main__":
+    unittest.main()
```
[the remark this answers is not in the document]
Comment (Ismatova-Rumiya) [reply solutions/tests/test_count_vowels.py:51]: Done!

**Review by Ismatova-Rumiya — COMMENTED** [review]
Taking all the comments into consideration I made the necessary changes.

**Review by Ismatova-Rumiya — COMMENTED** [review]
@JawidMohseni Taking your comments into consideration I edited the code

**Review by JawidMohseni — APPROVED** [review]
The code looks fine
